src/ts/components/product/Product.ts

```diff
@@ -0,0 +1,141 @@
+    addEvent(this.productTable, "click", this.handleEditProduct);
+    addEvent(this.productTable, "click", this.handleConfirmProduct);
+
+    const [productNameInput] = this.productInfoInputs;
+    productNameInput.focus();
+  }
+
+  handleAddProduct = (e: Event) => {
+    e.preventDefault();
+    this.productNameTdList = selectDomAll(".product-name", this.productTable);
+    const [productName, productPrice, productQuantity] = 
+      this.productInfoInputs.map((input: HTMLInputElement ) => input.value);
+    const productNameList = 
+      this.productNameTdList.map((productNameTd: HTMLTableCellElement) => productNameTd.textContent);
+      
+    try {
+      verifyProductInfo(
+        productName,
+        +productPrice,
+        +productQuantity,
+        productNameList
+      );
+      this.productInfoInputs.forEach((input: HTMLInputElement) => (input.value = ""));
+      const [productNameInput] = this.productInfoInputs;
+      productNameInput.focus();
+    } catch ({ message }) {
+      alert(message);
+      return;
+    }
+
+    this.productTable.insertAdjacentHTML(
+      "beforeend",
+      addProductTemplate(productName, +productPrice, +productQuantity)
+    );
+  };
+
+  handleRemoveProduct = (e: { target: HTMLTableElement }) => {
+    if (!e.target.classList.contains("product-remove-button")) {
+      return;
+    };
+    if (!confirm("정말 삭제하시겠습니까?")) {
+      return;
+    };
+
+    e.target.closest("tr").remove();
+  };
+
+  handleEditProduct = (e: { target: HTMLTableElement }) => {
+    if (!e.target.classList.contains("product-edit-button")) {
+      return;
+    }
+
+    const [productNameTd, productPriceTd, productQuantityTd] = 
+      Array.from(e.target.closest("tr").children);
+
+    e.target.closest("tr").innerHTML = editProductTemplate(
+      productNameTd.textContent,
+      +productPriceTd.textContent,
+      +productQuantityTd.textContent
+    );
+
+    const [productEditInput] = selectDomAll(".product-edit-input");
+    productEditInput.focus();
+  };
+
+  handleConfirmProduct = (e: { target: HTMLTableElement }) => {
+    if (!e.target.classList.contains("product-confirm-button")) {
+      return;
+    }
+
+    const [productName, productPrice, productQuantity] = Array.from(
+      selectDomAll(".product-edit-input"),
+      (input: HTMLInputElement) => input.value
+    );
+    const productNameList =
+      this.productNameTdList.map((productNameTd: HTMLTableCellElement) => productNameTd.textContent)
+
+    try {
+      verifyProductInfo(
+        productName,
+        +productPrice,
+        +productQuantity,
+        productNameList
+      );
+      this.changeEditProductInfo(
+        productName,
+        +productPrice,
+        +productQuantity,
+        e.target
+      );
+    } catch ({ message }) {
+      alert(message);
+    }
+  };
+
+  changeEditProductInfo = (
+    productName: string,
+    productPrice: number,
+    productQuantity: number,
+    target: HTMLElement
+  ) => {
```

Comment:
```suggestion
  changeEditProductInfo = ({
    productName,
    productPrice,
    productQuantity,
    target,
  }: {
    productName: string;
    productPrice: number;
    productQuantity: number;
    target: HTMLElement;
  }) => {
```

아니면

```suggestion
  changeEditProductInfo = ({
    productName,
    productPrice,
    productQuantity,
    target
  }: ChangeEditProductInfoProps) => {
```
이렇게 객체 형식으로 묶어줍시다~
[참고](https://github.com/qkraudghgh/clean-code-javascript-ko#%ED%95%A8%EC%88%98-%EC%9D%B8%EC%9E%90%EB%8A%94-2%EA%B0%9C-%EC%9D%B4%ED%95%98%EA%B0%80-%EC%9D%B4%EC%83%81%EC%A0%81%EC%9E%85%EB%8B%88%EB%8B%A4)